vars/transportRequestCreate.groovy

```diff
@@ -0,0 +1,68 @@
+import com.sap.piper.GitUtils
+import groovy.transform.Field
+
+import com.sap.piper.ConfigurationMerger
+import com.sap.piper.cm.ChangeManagement
+import com.sap.piper.cm.ChangeManagementException
+
+import hudson.AbortException
+
+
+@Field def STEP_NAME = 'transportRequestCreate'
+
+@Field Set parameterKeys = [
+    'changeId',
+    'developmentSystemId',
+    'cmCredentialsId',
+    'cmEndpoint'
+  ]
+
+@Field Set stepConfigurationKeys = [
+    'cmCredentialsId',
+    'cmEndpoint'
+  ]
+
+def call(parameters = [:]) {
+
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+
+        def script = parameters?.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
+
+        ChangeManagement cm = new ChangeManagement(script)
+
+        Map configuration = ConfigurationMerger.merge(parameters.script, STEP_NAME,
+                                                      parameters, parameterKeys,
+                                                      stepConfigurationKeys)
+
+        def changeId = configuration.changeId
+        if(!changeId) throw new AbortException('Change id not provided (parameter: \'changeId\').')
+
+        def developmentSystemId = configuration.developmentSystemId
```

Comment:
Why not having the sanity checks inside the ChangeManagementUtiliy class inside method ```createTransportRequest```. With that approach we would have a check regardless from where the method gets called.

Reply:
I think we always do it in the step after the loading the configuration because then you know it is a configuration problem, if we do it in the utils class, it could have been lost in between, and this would be an implementation error, that it is important for the developer, but not for the user.